### PR TITLE
Update TypeScript to 3.8.3 and Prettier to 2.0.4

### PR DESCRIPTION
**Pull request by NaridaL**
I want to use ?. in my other pull request.

### CODE REVIEW

**Comment by NaridaL** [comment]
@xaviergonz gonna tag you because you were active here recently 😄

**Comment by mweststrate** [comment]
Please do make sure you don't commit unrelated files. Your prettier configs seems to deviate from what is used in this repo, making this PR much larger than it should be. Feel free to commit a `.prettierrc` file to fix this if this avoids this repo conflicting with your IDE defaults.

File: publish.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const readline = require("readline")
 
 const rl = readline.createInterface({
     input: process.stdin,
-    output: process.stdout
+    output: process.stdout,
```
Comment (mweststrate) [inline publish.js:11]: This seems to be the result of using a different prettier config or version. Please align your local settings with this repo, or don't stage the unrelated files, or creating a PR that locks the config and prettier version would be even more awsome!

**Comment by NaridaL** [comment]
Yeah sorry, this PR should be called "update typescript and prettier". Prettier 2 supports the TS 3.8 features.

Prettier 2 contains some breaking changes concerning default options and formatting, hence the number of changes. See also https://prettier.io/blog/2020/03/21/2.0.0.html

As the new defaults make sense I thought I'd just reformat everything although I guess we could manually change the config so it doesn't change much.

**Comment by mweststrate** [comment]
@NaridaL  that makes sense, thanks for the explanation! The new dangling commas everywhere, is that a change in the prettier defaults? If it isn't, the .prettierrc should be updated accordingly

**Comment by mweststrate** [comment]
Answer: yes, it is the new default

**Comment by NaridaL** [comment]
@mweststrate do you mind hitting merge? Or you can add me as maintainer here too.

Also, I'd appreciate you taking a look at #245 😄

**Comment by mweststrate** [comment]
@NaridaL sorry, very swamped atm! I'll send an invite.
